habitat_sim/utils/datasets_download.py

```diff
@@ -0,0 +1,246 @@
+for key in data_groups:
+    assert key not in data_sources, "Duplicate key: " + key
+
+
+def prompt_yes_no(message):
+    print("\n-------------------------")
+    print(message)
+    while True:
+        answer = input("(y|n): ")
+        if answer.lower() == "y":
+            return True
+        elif answer.lower() == "n":
+            return False
+        else:
+            print("Invalid answer...")
+
+
+def clean_data(uid):
+    if uid not in data_sources:
+        print("Data download failed, no datasource named " + uid)
+        return
+    print(
+        "Cleaning datasource " + uid + ' directory: "' + data_sources[uid]["root"] + '"'
+    )
+    os.system("rm -r " + data_sources[uid]["root"])
+
+
+def download_and_place(uid, replace=False):
+    if uid not in data_sources:
+        print("Data download failed, no datasource named " + uid)
+        return
+
+    # check for current version
+    version_filepath = os.path.join(
+        data_dir,
+        data_sources[uid]["root"] + "DATA_VERSION_" + data_sources[uid]["version"],
+    )
+    if os.path.exists(data_sources[uid]["root"]):
+        resolved_existing = False
+        replace_existing = False
+        for file in os.listdir(data_sources[uid]["root"]):
+            if "DATA_VERSION" in file:
+                existing_version = file[13:]
+                if existing_version == data_sources[uid]["version"]:
+                    print(
+                        "Existing data source ("
+                        + uid
+                        + ") version ("
+                        + existing_version
+                        + ") is current, aborting download."
+                        + " Root directory: "
+                        + data_sources[uid]["root"]
+                    )
+                    return
+                else:
+                    # found root directory with different version
+                    found_version_message = (
+                        "("
+                        + uid
+                        + ") Found previous data version ("
+                        + existing_version
+                        + "). Replace with requested version ("
+                        + data_sources[uid]["version"]
+                        + ")?"
+                    )
+                    replace_existing = (
+                        replace if replace else prompt_yes_no(found_version_message)
+                    )
+                    resolved_existing = True
+        if not resolved_existing:
+            # found root directory with unspecified version
+            found_version_message = (
+                "("
+                + uid
+                + ") Found unknown data version. Replace with requested version ("
+                + data_sources[uid]["version"]
+                + ")?"
+            )
+            replace_existing = (
+                replace if replace else prompt_yes_no(found_version_message)
```

Comment:
invalid answer doesn't error or return. I would make it loop until they pick yes / no.

Reply:
The prompt_yes_no function loops until a valid input is provided.